src/OrleansRuntime/Storage/ProviderManagerBase.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Orleans.Providers;
+using Orleans.Runtime.Configuration;
+using Orleans.Storage;
+
+namespace Orleans.Runtime.Storage
+{
+    internal abstract class ProviderManagerBase<TProvider> : IProviderManagerBase<TProvider> where TProvider : IProvider
+    {
+        protected readonly IProviderRuntime providerRuntime;
+        protected ProviderLoader<TProvider> providerLoader;
+
+        public ProviderManagerBase(IGrainFactory grainFactory, IServiceProvider serviceProvider, IProviderRuntime providerRuntime)
+        {
+            this.providerRuntime = providerRuntime;
+            GrainFactory = grainFactory;
+            ServiceProvider = serviceProvider;
+        }
+
+        public abstract string ProviderCategory { get; }
+
+        internal Task LoadProviders(IDictionary<string, ProviderCategoryConfiguration> configs)
+        {
+            providerLoader = new ProviderLoader<TProvider>();
```

Comment:
Would it make sense to move this to the ctor? Maybe there's some reason to defer it which I'm not seeing yet.